examples/step-7/step-7.cc

```diff
@@ -998,19 +1000,16 @@ namespace Step7
             GridGenerator::hyper_cube(triangulation, -1, 1);
             triangulation.refine_global(3);
 
-            typename Triangulation<dim>::cell_iterator cell =
-                                                         triangulation.begin(),
-                                                       endc =
-                                                         triangulation.end();
-            for (; cell != endc; ++cell)
+            for (const auto cell : triangulation.cell_iterators())
               for (unsigned int face_number = 0;
                    face_number < GeometryInfo<dim>::faces_per_cell;
                    ++face_number)
-                if ((std::fabs(cell->face(face_number)->center()(0) - (-1)) <
-                     1e-12) ||
-                    (std::fabs(cell->face(face_number)->center()(1) - (-1)) <
-                     1e-12))
-                  cell->face(face_number)->set_boundary_id(1);
+                {
+                  const auto center = cell->face(face_number)->center();
+                  if ((std::fabs(center(0) - (-1)) < 1e-12) ||
+                      (std::fabs(center(1) - (-1)) < 1e-12))
+                    cell->face(face_number)->set_boundary_id(1);
```

Comment:
Question: I like this idom of creating a `const auto & ... = ...` because it creates easily readable code. Shall we keep this version?

Reply:
Updated to `const auto center =`